Assets/Scripts/Door.cs

```diff
@@ -62,22 +90,18 @@ private void Awake()
     void Start()
     {
         animator = GetComponent<Animator>();
-        animator.SetInteger("Type", doorType);
+        animator.SetInteger(Animator.StringToHash(animTypeBool),
+            (int)doorType);
+        audioSource = GetComponent<AudioSource>();
     }
 
-    /// <summary>
-    /// Add to the current solution
-    /// </summary>
-    /// <param name="entry">The solution piece connected to the activated switch</param>
+    // Functionality matches that described in IPuzzle
     public void AppendSolution(char entry)
     {
         CurrentSolution += entry;
     }
 
-    /// <summary>
-    /// Adds a switch to the list of connected switches
-    /// </summary>
-    /// <param name="addedSwitch">The switch to add</param>
+    // Functionality matches that described in IPuzzle
     public void AppendSwitchList(Switch addedSwitch)
     {
         connectedSwitches.Add(addedSwitch);
```

Comment:
make a property that encapsulates connected switches and allow people to add to it that way.